### PR TITLE
Add Maestro scenarios list and validation details

### PR DESCRIPTION
**Pull request by riarenas**
Struggled somewhat with this doc. Let me know where we want more / less details

### CODE REVIEW

File: src/Maestro/tests/scenarios.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Maestro++ Scenarios and test coverage
+
+## Overview
+
+This is a list of the business scenarios that Maestro is involved in, and the automated or manual validation to be performed for those scenarios
+
+## Package Publishing
+
+Maestro is only involved in the insertion of builds into the BAR.
```
Comment (markwilkie) [inline src/Maestro/tests/scenarios.md:9]: Does this mean that there's no scenario to test?  Not sure how to interpret the statement

File: src/Maestro/tests/scenarios.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+### Automated Tests
+Scenario tests for the latest version of the API are available at https://github.com/dotnet/arcade-services/blob/master/src/Maestro/tests/Scenarios/builds.ps1
+
+Unit tests for the 20190116 API version are available at: https://github.com/dotnet/arcade-services/blob/master/src/Maestro/Maestro.Web.Tests/BuildController20190116Tests.cs
+
+## Manual testing
```
Comment (markwilkie) [inline src/Maestro/tests/scenarios.md:16]: So what are the manual tests we should run?  Just the two post statement below?

**Comment by sunandabalu** [comment]
This is a good start 👍

File: src/Maestro/tests/scenarios.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    }
+  ],
+  "dependencies": [
+  ]
+}
+```
+
+Test that it's possible to publish a build with assets to the BAR for versions not covered by the automated scenario tests.
+
+## Dependency Flow
```
Comment (markwilkie) [inline src/Maestro/tests/scenarios.md:48]: Is there a delta?  I think the hope is that with this doc we can not only understand what is currently being tested, but what needs to be manually tested (and automated in the future).  

In other words, I don't see any manual test section here, does that imply we're good?

File: src/Maestro/tests/scenarios.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+We have some automated UI testing in <https://github.com/dotnet/arcade-services/tree/master/src/Maestro/maestro-angular/e2e>, but it has failed to detect the latest issues we've seen.
+
+We have also started adding unit tests for data retrieval from the Database that should be expanded as new regressions are encountered: https://github.com/dotnet/arcade-services/tree/master/src/Maestro/Maestro.Web.Tests
+
+### Manual Testing
+
+The brunt of Barviz validation is performed manually:
+
+* Go through the website making sure no errors are rendered when browsing channels / repositories and builds:
```
Comment (markwilkie) [inline src/Maestro/tests/scenarios.md:97]: Probably a quick pass at which scenarios to look at.  It's probably hard for most folks to "go through the website" grin...  Nothing extensive, but a quick pass at it is probably good.

Maybe that's what the following sections are supposed to do?  (as you can see, I'm having a little bit of a hard time following the doc - but I think that's mostly on me)

File: src/Maestro/tests/scenarios.md (hunk 1 above)
Comment (riarenas) [reply src/Maestro/tests/scenarios.md:9]: Yeah, basically struggled with conveying a conclusion about every section. IN this case, I meant that we only care about build creation, and that is tested both by the scenario test that I mention, and the post requests to the API. I'll reword

File: src/Maestro/tests/scenarios.md (hunk 2 above)
Comment (riarenas) [reply src/Maestro/tests/scenarios.md:16]: Yes. If creating a build works, then publishing should work.

File: src/Maestro/tests/scenarios.md (hunk 3 above)
Comment (riarenas) [reply src/Maestro/tests/scenarios.md:48]: Yes, from my perspective we're good with the scenario tests when it comes to dependency flow

File: src/Maestro/tests/scenarios.md (hunk 4 above)
Comment (riarenas) [reply src/Maestro/tests/scenarios.md:97]: Added some details on how the two UI views look like

**Pull request merged**